src/tools/cephfs/Dumper.cc

```diff
@@ -226,24 +226,50 @@ int Dumper::undump(const char *dump_file, bool force)
 
   char buf[HEADER_LEN];
   r = safe_read(fd, buf, sizeof(buf));
-  if (r < 0) {
+  if (r <= 0) {
```

Comment:
Do you have a better solution?

Reply:
i can think of something like treating the empty read a special case i.e. have a separate derr for it saying "Empty journal file" or similar. like:
```
if (r <= 0) {
	if (r == 0) {
		derr << "No-op since empty journal file " << dump_file << std::endl;
	} else {
		derr << "Error reading " << dump_file << std::endl;
	}
	VOID_TEMP_FAILURE_RETRY(::close(fd));
	return r;
}
```